src/itensor.jl

```diff
@@ -52,8 +58,78 @@ function ITensor(A::Array{S},inds::IndexSet) where {S<:Number}
 end
 ITensor(A::Array{S},inds::Index...) where {S<:Number} = ITensor(A,IndexSet(inds...))
 
-# Convert to complex
-complex(T::ITensor) = ITensor(inds(T),storage_complex(store(T)))
+#
+# Diag ITensor constructors
+#
+
+"""
+diagITensor(::Type{T}, is::IndexSet)
+
+Make an ITensor with Diag storage with element type T.
```

Comment:
What do you think of saying:
"Make a sparse ITensor with non-zero elements only along the diagonal. The storage will have Diag type."
...or similar. Just to make it friendlier to newcomers who may not appreciate what Diag storage does.

Reply:
Yeah that sounds good, I can change it.